x-pack/elastic-agent/pkg/composable/providers/env/env.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package env
+
+import (
+	"os"
+	"strings"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+)
+
+func init() {
+	composable.Providers.AddContextProvider("env", ContextProviderBuilder)
+}
+
+type contextProvider struct{}
+
+// Run runs the environment context provider.
+func (*contextProvider) Run(comm composable.ContextProviderComm) error {
+	comm.Set(getEnvMapping())
+	return nil
+}
+
+// ContextProviderBuilder builds the context provider.
+func ContextProviderBuilder(_ *config.Config) (composable.ContextProvider, error) {
+	return &contextProvider{}, nil
+}
+
+func getEnvMapping() map[string]interface{} {
```

Comment:
I wonder if at one stage it will matter if the value is a `string` or `long` as some of these might be added to the mapping of the event. In general, I think it should be the providers job to provide the correct types?

Reply:
Key/values from ENV in golang are always strings. You can convert the string to an integer or float manually.

The good thing is that even if the value is a number, it will be read in as a string, substituted just fine in the configuration and passed onto Filebeat, Metricbeat, etc.